control-plane/pkg/reconciler/broker/broker_test.go

```diff
@@ -2005,6 +2020,56 @@ func brokerFinalization(t *testing.T, format string, env config.Env) {
 				testProber: probertesting.MockProber(prober.StatusNotReady),
 			},
 		},
+		{
+			Name: "Reconciled normal - with auth config",
+			Objects: []runtime.Object{
+				NewDeletedBroker(
+					WithExternalTopic(ExternalTopicName),
+					WithBrokerConfig(KReference(BrokerConfig(bootstrapServers, 20, 5,
+						BrokerAuthConfig("secret-1"),
+					))),
+					BrokerConfigMapSecretAnnotation("secret-1"),
+				),
+				NewSSLSecret(ConfigMapNamespace, "secret-1"),
```

Comment:
Can we add a finalizer to the secret so that we test that the finalizer is properly removed?